src/operator/subgraph/mkldnn/mkldnn_conv.h

```diff
@@ -0,0 +1,274 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef MXNET_OPERATOR_SUBGRAPH_MKLDNN_CONV_H_
+#define MXNET_OPERATOR_SUBGRAPH_MKLDNN_CONV_H_
+
+#if MXNET_USE_MKLDNN == 1
+
+#include "../common.h"
+#include "../subgraph_property.h"
+#include "../../nn/convolution-inl.h"
+#include "../../nn/activation-inl.h"
+
+namespace mxnet {
+namespace op {
+namespace sg {
+
+class SgMKLDNNConvSelector : public SubgraphSelector {
+ public:
+  /*! \brief pattern match status */
+  enum SelectStatus {
```

Comment:
It's better to use [typed enums](https://en.cppreference.com/w/cpp/language/enum) since they have better scoping and you don't need any prefixes or all caps. Usage will be like: `SelectStatus.Fail`